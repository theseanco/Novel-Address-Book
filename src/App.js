import React, { useState } from 'react';
import Footer from './components/Footer/Footer';
import Header from './components/Header/Header';
import AddressList from './components/AddressList/AddressList';
import AddressForm from './components/AddAddress/AddressForm/AddressForm';

function App() {
  // State for the app that is used to set whether fetching should be performed or not, a boolean
  // This could be done using Redux, which would make the relationship between components less heirarchical
  // But in this case, for a small prototype with everything else being handled by local state Redux is (arguably) overkill
  // This could also use the context API, but again that's probably overkill for a boolean value.
  // When an item is successfully posted to the database, this will be set to `true` by AddressForm
  // This will trigger a refetch and rerender in AddressList, which will then set it to false. 
  const [fetching, setFetching] = useState(true);

  return (
    <>
      <Header />
      <AddressList fetching={fetching} setFetching={setFetching}/>
      <AddressForm setFetching={setFetching} />
      <Footer />
    </>
  );
}

export default App;
