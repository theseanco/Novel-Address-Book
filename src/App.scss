@import "./variables.scss";

// Used to set rem and em for sizing: see https://css-tricks.com/rems-ems/
html {
  font-size: 15px;

  body {
    background-color: $black;
    margin: 0px;
    padding: 0px 2rem;

    #root {
      display: flex;
      flex-direction: column;
      min-height: 100vh;
    }

    main {
      background-color: $black;
      flex: 1;
      padding: $masterPadding;
    }

    header {
      border-bottom: 1px solid $dark;

      h1 {
        color: $headerfooter;
        margin: 1rem;
      }
    }

    footer {
      border-top : 1px solid $dark;

      h4 {
        color: $headerfooter;
        margin: 1rem;
      }
    }

    header, footer {
      background-color: $black;
      padding-top: 2rem;
      padding: $masterPadding;
    }
  }
}



