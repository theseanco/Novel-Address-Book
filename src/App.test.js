import React from 'react';
import { render, getByTestId, getByText } from '@testing-library/react';
import App from './App';

describe('App Root tests', () => {

  it('Should contain a header and a footer', () => {
    const { container } = render(<App />);
    expect(getByTestId(container, 'header')).toBeInTheDocument();
    expect(getByTestId(container, 'footer')).toBeInTheDocument();
    // Check the list is loading
    expect(getByText(container, /Loading addresses/)).toBeInTheDocument();
    // Check the form has rendered
    expect(getByText(container, /Add Address/)).toBeInTheDocument();
  })

  // Snapshot test for the app component as it does not have stateful UI logic
  it('should match snapshot', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  })
});