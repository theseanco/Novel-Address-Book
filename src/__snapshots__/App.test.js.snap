// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Root tests should match snapshot 1`] = `
<header
  data-testid="header"
>
  <h1>
    Address Book App
  </h1>
</header>
`;
